src/main/java/hudson/remoting/RemoteClassLoader.java

```diff
@@ -105,14 +109,29 @@
      */
     private final Map<String,ClassReference> prefetchedClasses = Collections.synchronizedMap(new HashMap<String,ClassReference>());
 
-    public static ClassLoader create(ClassLoader parent, IClassLoader proxy) {
+    public static ClassLoader create(final ClassLoader parent, final IClassLoader proxy) {
         if(proxy instanceof ClassLoaderProxy) {
             // when the remote sends 'RemoteIClassLoader' as the proxy, on this side we get it
             // as ClassLoaderProxy. This means, the so-called remote classloader here is
             // actually our classloader that we exported to the other side.
             return ((ClassLoaderProxy)proxy).cl;
         }
-        return new RemoteClassLoader(parent, proxy);
+
+        // Create new RemoteClassLoader otherwise
+        final RemoteClassLoader created;
+        try {
+            created = AccessController.doPrivileged(
```

Comment:
You should only ever need to use this in case you are running with a security manager enabled, which for Remoting we are not. So I do not see the point of this PR. If it is just to get SpotBugs to shut up, use the warning suppression annotation, or just configure a SpotBugs exclude for that warning in the POM.

Reply:
And there is a Java JEP proposing to deprecate the security manager, and associated methods like this.